src/redux/categories/categoriesSlice.js

```diff
@@ -0,0 +1,18 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  categories: [],
+};
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+    checkStatus: (state) => {
+      state.categories = ['Under construction'];
+    },
+  },
+});
```

Comment:
- [ ] Great work so far! However, according to the [project requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions_v2.md), the reducer inside the categories slice needs to check the status before updating the state. It appears that this step has been missed in your code, so please ensure that you check the payload from the action before making any updates to the state, Here is an example of how to do this:
```jsx
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories

```
- [ ] As you might have guessed this also means you'll have to add a second parameter called `action` to the  `checkStatus` reducer.

Reply:
Thank you so much for your valuable feedback @AbelG101 , I really appreciate it.  I will make the changes according to that.